spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java

```diff
@@ -629,4 +627,36 @@ protected SSLContext createSSLContext() throws NoSuchAlgorithmException {
 		return SSLContext.getInstance(this.sslAlgorithm);
 	}
 
+
+	/**
+	 * In case the client doesn't provide the sslProperties location or the truststore is null delegate to Java's default
+	 * trust mechanism
+	 * @throws NoSuchAlgorithmException
+	 * @throws KeyManagementException
+	 * @throws KeyStoreException
+     */
+	private void useDefaultTrustStoreMechanism() throws NoSuchAlgorithmException, KeyManagementException,
+			KeyStoreException {
+		SSLContext c = SSLContext.getInstance(this.sslAlgorithm);
+		c.init(null, new TrustManager[] {  getDefaultTrustManager(TrustManagerFactory.getDefaultAlgorithm(), null) }, null);
+		this.connectionFactory.useSslProtocol(c);
+	}
+
+
+	private static X509TrustManager getDefaultTrustManager(String algorithm, KeyStore keystore) throws NoSuchAlgorithmException, KeyStoreException {
+
+		TrustManagerFactory factory = TrustManagerFactory.getInstance(algorithm);
+		X509TrustManager x509TrustManager = null;
+        factory.init(keystore);
```

Comment:
I don't know the answer persumably it might be a glue code for different  JDK implementations.

while searching; I did come across one article which showed on how to create sslcontext for IBM  jdks.
the doc says  `IBM TrustManagerFactory is called IbmX509 or IbmPKIX.` 
I presume spring classes has to work on any JDK.

http://www.ibm.com/support/knowledgecenter/SSYKE2_6.0.0/com.ibm.java.security.component.60.doc/security-component/jsse2Docs/knowndiffsun.html
http://www.ibm.com/support/knowledgecenter/SSYKE2_6.0.0/com.ibm.java.security.component.60.doc/security-component/jsse2Docs/pkixtrustmgrsupport.html